code/game/objects/items/ego_weapons/teth.dm

```diff
@@ -540,3 +540,26 @@
 		var/whack_speed = (prob(60) ? 1 : 4)
 		target.throw_at(throw_target, rand(1, 2), whack_speed, user)
 
+
+/obj/item/ego_weapon/lance/visions
```

Comment:
This is cool, although I am concerned that it might be a bit feature creeped for a beginner weapon.